Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -698,13 +705,43 @@ - (NSIndexPath *_Nullable)indexPathOfUnreadMessagesIndicator
     return nil;
 }
 
+- (NSIndexPath *_Nullable)indexPathOfMessageOnOpen
+{
```

Comment:
I was expecting something shorter here. We *have* the mapping, view index, and the group, so getting the index path should be a one liner:

- (nullable NSIndexPath *)indexPathForIndex:(NSUInteger)index inGroup:(NSString *)group;

But it would entail storing the viewIndex in addition to (or rather than) the reverse-index. In general I'm not sure introducing the concept of the "reverse index" is helpful. 

Instead we could store the forward index, which simplifies the method which stores it, use it here directly, and only invert it for calculating the range length.

You've already addressed my blocking perf concerns though - so this is TIOLI, I'm just trying to find ways to simplify the remaining code since we'll live with it forever.

Reply:
> But it would entail storing the viewIndex in addition to (or rather than) the reverse-index. In general I'm not sure introducing the concept of the "reverse index" is helpful.

We use "reverse indices" because (among other uses) we use this to determine the initial load window size, before a "view index" has any meaning.  Sorry if that wasn't clear; I've elaborated the comments.